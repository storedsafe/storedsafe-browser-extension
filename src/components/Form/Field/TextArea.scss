@import '../../../variables';

.field-textarea .field-control {
  border: 2px solid $border;
  border-radius: $border-radius;
}
