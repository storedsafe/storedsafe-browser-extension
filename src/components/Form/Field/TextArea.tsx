import * as React from 'react';
import { FieldProps, FieldAttributes } from './Field';
import './TextArea.scss';

type TextAreaProps = Pick<FieldProps, 'name' | 'value' | 'onChange' | 'onBlur' | 'label'>;
export const TextArea: React.SFC<TextAreaProps | FieldAttributes> = ({
  name,
  value,
  label,
  onChange,
  onBlur,
  ...attributes
}: TextAreaProps) => (
  <div className="field field-textarea">
    <label htmlFor={name}>
      <span className="form-label">{label}</span>
      <div className="field-container">
        <textarea
          className="field-control"
          id={name}
          name={name}
          value={value as string}
          onChange={({ target }: React.ChangeEvent<HTMLTextAreaElement>): void => {
            onChange && onChange(target.value, name);
          }}
          onBlur={({ target }: React.FocusEvent<HTMLTextAreaElement>): void => {
            onBlur && onBlur(target.value, name);
          }}
          {...attributes}
        />
      </div>
    </label>
  </div>
);
