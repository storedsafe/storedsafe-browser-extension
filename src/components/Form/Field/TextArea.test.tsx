import * as React from 'react';
import { TextArea } from './TextArea';
import { shallow, mount } from 'enzyme';

const onChange = jest.fn();
const onBlur = jest.fn();

beforeEach(() => {
  onChange.mockClear();
  onBlur.mockClear();
});

test('<TextArea />', () => {
  const wrapper = shallow(<TextArea name="test" label="test" />);
  expect(wrapper.find('textarea#test').length).toBe(1);
  expect(wrapper).toMatchSnapshot();
});

test('<TextArea />, change', () => {
  const value = 'This is a textarea';
  const wrapper = mount(<TextArea name="test" label="test" value={value} onChange={onChange} />);
  wrapper.find('textarea#test').simulate('change', { target: { name: 'test', value: 'mockChange' } });
  expect(onChange).toHaveBeenCalledWith('mockChange', 'test');
});

test('<TextArea />, blur', () => {
  const value = 'This is a textarea';
  const wrapper = mount(<TextArea name="test" label="test" value={value} onBlur={onBlur} />);
  wrapper.find('textarea#test').simulate('blur', { target: { name: 'test', value: 'mockBlur' } });
  expect(onBlur).toHaveBeenCalledWith('mockBlur', 'test');
});
