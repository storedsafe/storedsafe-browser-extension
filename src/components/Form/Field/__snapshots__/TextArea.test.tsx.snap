// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> 1`] = `ShallowWrapper {}`;
